### ISSUE TITLE
Move generated BNFC code to its own package

### ISSUE THREAD

**Issue by wenkokke** [vehicle-lang/vehicle#17]

**Comment by MatthewDaggitt** [comment]
Nice to have but this is blocked by the fact that `stack` doesn't seem to support internal libraries. See https://github.com/haskell/haskell-language-server/issues/2276 for details.

**Comment by MatthewDaggitt** [comment]
No longer blocked as we switched from Stack to Cabal
